core/src/main/java/org/elasticsearch/indices/IndicesModule.java

```diff
@@ -52,8 +65,15 @@
     private final ExtensionPoint.ClassSet<QueryParser> queryParsers
         = new ExtensionPoint.ClassSet<>("query_parser", QueryParser.class);
 
+    private final Map<String, Mapper.TypeParser> mapperParsers
```

Comment:
Maybe leave a message about why iteration order matters here because this is where the reader of this class first sees that it might.